CMakeLists.txt

```diff
@@ -97,6 +97,8 @@ function(set_platform_files c_util_dir)
         set(UNIQUEID_C_FILE ${c_util_dir}/adapters/uniqueid_win32.c PARENT_SCOPE)
         set(STRING_UTILS_C_FILE ${c_util_dir}/adapters/string_utils.c PARENT_SCOPE)
         set(SRW_LOCK_C_FILE ${c_util_dir}/adapters/srw_lock.c PARENT_SCOPE)
+        set(SM_C_FILE ${c_util_dir}/src/sm.c PARENT_SCOPE)
```

Comment:
>sm [](start = 40, length = 2)

I know it's poetry, but let's name this state_manager. I like the manager part in it :-). It will be easier for someone else (Parth, Cesar, future generations) looking at our code to figure out what this unit does.